sample_metadata/parser/cloudhelper.py

```diff
@@ -108,20 +109,20 @@ def populate_filename_map(
         so let's prepopulate that filename_map from the search_locations!
         """
 
-        fn_map = {}
         for directory in search_locations:
```

Comment:
I made this change because it would mean overriding this parser means you shouldn't have to know about state (and makes it much easier to mock) in tests. Curious about the errors you're seeing here though.

Reply:
Line 119, 
```
logging.warning(
                        f'File "{file}" from "{directory}" already exists in directory map: {self.filename_map[file_base]}'
)
```
References `self.filename_map` which is not initialised. Alternatively we could make the method static.

Reply:
Ah sorry I see, yes please if you could make it static that would be great.

Reply:
What about `self.list_directory(directory)` call? It would involve more refactoring.

Reply:
Btw, feel free to commit to the PR!

Reply:
I guess can make it pseudo-static, keeping the `self.list_directory` call